File: internship_cs_mar23/jury_solutions/cpp/b.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;

signed main() {
	int n;
	cin >> n;
	vector<pair<int, int>> f(n);
	set<pair<int, int>> s;
	for (int i = 0; i < n; ++i) {
		cin >> f[i].first >> f[i].second;
		s.insert(f[i]);
	}
	long long ans = 0;
	for (int i = 0; i < n; ++i) {
		for (int j = i + 1; j < n; ++j) {
			long long area = (long long)abs(f[i].first - f[j].first) *
				abs(f[i].second - f[j].second);
			if (area <= ans) continue;
			pair<int, int> a(f[i].first, f[j].second);
			pair<int, int> b(f[j].first, f[i].second);
			if (s.count(a) && s.count(b))
				ans = area;
		}
	}
	cout << ans;
}

File: internship_cs_mar23/jury_solutions/python/b.py
n = int(input())
f = [tuple(map(int, input().split())) for _ in range(n)]
s = set(f)
ans = 0
for i in range(n):
    for j in range(i + 1, n):
        area = abs(f[i][0] - f[j][0]) * abs(f[i][1] - f[j][1])
        if area <= ans: continue
        if (f[i][0], f[j][1]) in s and (f[j][0], f[i][1]) in s:
            ans = area
print(ans)
